src/lib/http/fetch.ts

```diff
@@ -1,4 +1,16 @@
-type RequestInit = Parameters<typeof fetch>[1];
+import type { RequestInitWithRetry } from 'fetch-retry';
+import fetchRetry from 'fetch-retry';
+import { fetch as nativeFetch } from 'undici';
+
+const delayMultiplier = process.env.NODE_ENV === 'test' ? 1 : 1000;
+
+const fetchAndRetry = fetchRetry(nativeFetch as any, {
+  retries: 5,
+  retryOn: [500, 501, 502, 503],
```

Comment:
maybe also 504 (gateway timeout) ?

Reply:
I purposely left that out. If you're getting a 504, that probably means the server is being DDOS'd or can't handle requests. Especially for internal services, if we are getting a 504 on permissions api then we want to just fail.